input/types/types.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"net"
+	"time"
+)
+
+// Builder specifes the interface any delimiter and parser must implement.
```

Comment:
`Builder` doesn't strike me as a natural term for this. I know naming is hard, but can you explain how you came up with Builder?